src/mscorlib/src/System/IO/StreamReader.cs

```diff
@@ -850,7 +869,7 @@ public override Task<String> ReadToEndAsync()
         private async Task<String> ReadToEndAsyncInternal()
         {
             // Call ReadBuffer, then pull data out of charBuffer.
-            StringBuilder sb = new StringBuilder(CharLen_Prop - CharPos_Prop);
+            StringBuilder sb = AcquireStringBuilder(CharLen_Prop - CharPos_Prop);
```

Comment:
Same as above. This is an async method. Is it OK to use the shared StringBuilder here?